### PR TITLE
Mas i1788 reconcileapi

### PR DESCRIPTION
**Pull request by martinsumner**
Changes for https://github.com/basho/riak_kv/pull/1792

### CODE REVIEW

File: rebar.config [hunk 1]
```diff
@@ -17,7 +17,7 @@
 ]}.
 
 {deps, [
-    {riak_pb, {git, "https://github.com/basho/riak_pb", {tag, "3.0.4"}}}
+    {riak_pb, {git, "https://github.com/basho/riak_pb", {branch, "mas-i1788-reconcileapi"}}}
```
Comment (ThomasArts) [inline rebar.config]: to do... point to right tag

File: src/riakc_pb_socket.erl [hunk 2]
```diff
@@ -338,15 +338,48 @@ get(Pid, Bucket, Key, Options, Timeout) ->
...
     call_infinity(Pid, {req, Req, Timeout}).
 
+
 %% @doc Fetch replicated objects from a queue
 -spec fetch(pid(), binary()) ->
                 {ok, queue_empty}|
-                {ok|crc_wonky, {deleted, term(), binary()}|binary()}.
+                {ok|crc_wonky,
+                    {deleted, term(), binary()}|binary()}.
 fetch(Pid, QueueName) ->
```
Comment (ThomasArts) [inline src/riakc_pb_socket.erl:347]: I would rather see that this function calls
```suggestion
fetch(Pid, QueueName) ->
   fetch(Pid, QueueName, internal).
```

then backward compatibility is more clear.

File: src/riakc_pb_socket.erl [hunk 3]
```diff
@@ -338,15 +338,48 @@ get(Pid, Bucket, Key, Options, Timeout) ->
...
+%% @doc Fetch with specific format may also return segment ID and hash
+-spec fetch(pid(), binary(), internal|internal_aaehash) ->
+                {ok, queue_empty}|
+                {ok|crc_wonky,
+                    {deleted, term(), binary()}|
+                        binary()|
+                        {deleted, term(), binary(), non_neg_integer(), non_neg_integer()}|
+                        {binary(), non_neg_integer(), non_neg_integer()}}.
+fetch(Pid, QueueName, ObjectFormat)
+        when ObjectFormat =:= internal; ObjectFormat =:= internal_aaehash->
```
Comment (ThomasArts) [inline src/riakc_pb_socket.erl]:
```suggestion
        when ObjectFormat =:= internal; ObjectFormat =:= internal_aaehash ->
```

File: src/riakc_pb_socket.erl [hunk 4]
```diff
@@ -2382,14 +2415,58 @@ process_response(#request{msg = #rpbfetchreq{}},
...
+        {ok, 
+            list_to_binary(
+                lists:flatten(
+                    io_lib:format("Queue ~s: ~w ~w ~w", [Q, FL, FSL, RTL])))},
+        State};
+process_response(#request{msg = #rpbpushreq{queuename = Q}}, 
+                    #rpbpushresp{queue_exists = false}, State) ->
+    {reply,
+        {ok, 
+            list_to_binary(
```
Comment (ThomasArts) [inline src/riakc_pb_socket.erl]:
```suggestion
            iolist_to_binary(
```
then you can leave out flatten... same in line 2456

**Review by ThomasArts — APPROVED** [review]

File: src/riakc_pb_socket.erl (hunk 2 above)
Comment (martinsumner) [reply src/riakc_pb_socket.erl:347]: Fixed

File: src/riakc_pb_socket.erl (hunk 4 above)
Comment (martinsumner) [reply src/riakc_pb_socket.erl]: Fixed

**Pull request merged**